subsys/usb/class/hid/core.c

```diff
@@ -123,11 +143,54 @@ static int hid_on_get_protocol(struct usb_setup_packet *setup, s32_t *len,
 static int hid_on_set_idle(struct usb_setup_packet *setup, s32_t *len,
 			   u8_t **data)
 {
-	LOG_DBG("Set Idle callback");
+#ifdef CONFIG_USB_DEVICE_SOF
+	u8_t rate = ((sys_le16_to_cpu(setup->wValue) & 0xFF00) >> 8);
+	u8_t report_id = sys_le16_to_cpu(setup->wValue) & 0xFF;
 
-	/* TODO: Do something. */
+	if (report_id > CONFIG_USB_HID_REPORTS) {
+		LOG_ERR("Report id out of limit: %d", report_id);
+		return -ENOTSUP;
+	}
+
+	LOG_DBG("Set Idle callback, rate: %d, report_id: %d", rate, report_id);
+
+	hid_device.idle_rate[report_id] = rate;
 
+	if (rate == 0) {
+		/* Clear idle */
+		bool clear = true;
+		u8_t i = 0;
```

Comment:
u8_t or u16_t ? [Below](https://github.com/zephyrproject-rtos/zephyr/pull/11802/commits/751df9599689aa7b20990582f308e8b1b03012f2#diff-ca310b6eb6831a8122b8134d5fdb6ae6R225) it is u16_t

Reply:
u16_t. Fixed.